pkgs/applications/office/paperless/default.nix

```diff
@@ -0,0 +1,168 @@
+# let
+#   pythonEnv = paperless.python.withPackages (ps: paperless.runtimePackages ++ [ ps.gunicorn ]);
+# in
+#   writers.writeBash "run-gunicorn" ''
+#     source ${paperless}/share/paperless/setup-env.sh
+#     PYTHONPATH=$paperlessSrc ${pythonEnv}/bin/gunicorn paperless.wsgi
+#   ''
+
+let
+  paperless = stdenv.mkDerivation rec {
+    name = "paperless-${version}";
+    version = "2.7.0";
+
+    src = fetchFromGitHub {
+      owner = "the-paperless-project";
+      repo = "paperless";
+      rev = version;
+      sha256 = "0pkmyky1crjnsg7r0gfk0fadisfsgzlsq6afpz16wx4hp6yvkkf7";
+    };
+
+    nativeBuildInputs = [ makeWrapper ];
+
+    doCheck = true;
+    dontInstall = true;
+
+    pythonEnv      = python.withPackages (_: runtimePackages);
+    pythonCheckEnv = python.withPackages (_: (runtimePackages ++ checkPackages));
+
+    unpackPhase = ''
+      srcDir=$out/share/paperless
+      mkdir -p $srcDir
+      cp -r --no-preserve=mode $src/src/* $src/LICENSE $srcDir
+    '';
+
+    buildPhase = let
+      # Paperless has explicit runtime checks that expect these binaries to be in PATH
+      extraBin = lib.makeBinPath [ imagemagick7 ghostscript optipng tesseract unpaper ];
+    in ''
+      ${python.interpreter} -m compileall $srcDir
+
+      makeWrapper $pythonEnv/bin/python $out/bin/paperless \
+        --set PATH ${extraBin} --add-flags $out/share/paperless/manage.py
+
+      # A shell snippet that can be sourced to setup a paperless env
+      cat > $out/share/paperless/setup-env.sh <<EOF
+      export PATH="$pythonEnv/bin:${extraBin}''${PATH:+:}$PATH"
+      export paperlessSrc=$out/share/paperless
+      EOF
+    '';
+
+    checkPhase = ''
+      source $out/share/paperless/setup-env.sh
+      tmpDir=$(realpath testsTmp)
+      mkdir $tmpDir
+      export HOME=$tmpDir
+      export PAPERLESS_MEDIADIR=$tmpDir
+      cd $paperlessSrc
+      # Prevent tests from writing to the derivation output
+      chmod -R -w $out
+      # Disable cache to silence a pytest warning ("could not create cache")
+      $pythonCheckEnv/bin/pytest -p no:cacheprovider
+    '';
+
+    passthru = {
+      withConfig = callPackage ./withConfig.nix {};
+      inherit python runtimePackages checkPackages tesseract;
+    };
+
+    meta = with lib; {
+      description = "Scan, index, and archive all of your paper documents";
+      homepage = https://github.com/the-paperless-project/paperless;
+      license = licenses.gpl3;
+      maintainers = [ maintainers.earvstedt ];
+    };
+  };
+
+  # Use Python 3.6 because dateparser is broken with 3.7 (https://github.com/NixOS/nixpkgs/issues/52766)
+  python = python36.override {
+    packageOverrides = self: super: {
+      # Paperless only supports Django 2.0
+      django = django_2_0 super;
+      pyocr = pyocrWithUserTesseract super;
+    };
+  };
+
+  django_2_0 = pyPkgs: pyPkgs.django_2_1.overrideDerivation (_: rec {
+    pname = "Django";
+    version = "2.0.12";
+    name = "${pname}-${version}";
```

Comment:
```suggestion
```

Reply:
For `overrideDerivation` , we have to update `name` manually.